core/arch/arm/mm/core_mmu.c

```diff
@@ -837,12 +843,14 @@ static void init_mem_map(struct tee_mmap_region *memory_map, size_t num_elems)
 	assert(va >= CFG_TEE_RAM_START);
 	assert(end <= CFG_TEE_RAM_START + CFG_TEE_RAM_VA_SIZE);
 
-	add_pager_vaspace(memory_map, num_elems, va, end, &last);
+	add_pager_vaspace(memory_map, num_elems, va, end, &last,
+			  &next_pgdir_va);
 
```

Comment:
why ? What insures the first area mapped right below TEE_RAM (case mapping setup with "tee ram at top")  is pgdir aligned ? At this stage, va may not be pgdir aligned.